packages/flutter/lib/src/painting/_network_image_web.dart

```diff
@@ -287,25 +272,14 @@ class _ForwardingImageStreamCompleter extends ImageStreamCompleter {
         if (_disposed) {
           // Add a listener since the delegate completer won't dispose if it never
           // had a listener.
-          value.addListener(ImageStreamListener((_, _) {}));
+          final ImageStreamListener listener = ImageStreamListener((_, _) {});
+          value.addListener(listener);
+          value.removeListener(listener);
```

Comment:
If it improves the situation, let's start with this, but there's some smell about having to add + remove a listener before disposing of it.

Reply:
I agree. I added a new method `markHadListener` to bypass this check. What do you think of this approach?

Reply:
With discussion we've solved this issue by removing this check. https://github.com/flutter/flutter/pull/166797